src/openrct2/peep/Guest.cpp

```diff
@@ -3728,9 +3728,18 @@ void Guest::UpdateRideAdvanceThroughEntrance()
     }
 
     Vehicle* vehicle = GET_VEHICLE(ride->vehicles[current_train]);
+    if (vehicle == nullptr)
+    {
+        return;
+    }
+
     for (int32_t i = current_car; i != 0; --i)
     {
         vehicle = GET_VEHICLE(vehicle->next_vehicle_on_train);
+        if (vehicle == nullptr)
```

Comment:
Should this one really have been added? I guess the for was to look for the vehicle, no?

Reply:
Just take a good look at the code. It keeps fetching the next vehicle, which could eventually be `NULL`, even before the for loop ends.